python/tk_nuke/__init__.py

```diff
@@ -120,21 +116,23 @@ def __tank_on_save_callback():
         # API instance.
         try:
             tk = tank.tank_from_path(file_name)
```

Comment:
Instead of trying to create a new `tank` object in any case, we could maybe check  if there is a current engine and take the associated `tank` object.

Like [here](https://github.com/shotgunsoftware/tk-nuke/pull/58/files#diff-2082bc816b93dbd02e442c7942fa7709R140)

Reply:
Thanks for the suggestions. I've spent a little bit of time looking at this, but I don't think we can reliably reuse the sgtk instance. If the new script file path is in a different project, then this will cause errors later.
If you do the following:
`sgtk.platform.current_engine().sgtk.context_from_path(file_name)`
it won't error (unlike `tk = tank.tank_from_path(file_name)`), but will return back a context containing only the site url, and then it will attempt to start the engine in site context, where as really it should be disabling the Shotgun integration.